packages/pkl.tar/examples/archive.pax.pkl

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// Test pax archives, emitting fields that are too large/long for UStar.
+amends "archive.pkl"
+
+import ".../File.pkl"
+
+local superLongPathPrefix = List("some/long/prefix").repeat(10).join("/")
+local superLongPathParent = List("../../..").repeat(10).join("/")
+
+archive {
+  ["\(superLongPathPrefix)/regularFile.txt"] {
+    text = "this is a regular file"
+  }
+  ["\(superLongPathPrefix)/someDir/regularFile.txt"] = this["regularFile.txt"]
+  ["\(superLongPathPrefix)/someDir/hardLink.txt"] = new File.Link {
+    target = "\(superLongPathParent)/regularFile.txt"
+  }
+  ["\(superLongPathPrefix)/someDir/symLink.txt"] = new File.SymbolicLink {
+    target = "\(superLongPathParent)/regularFile.txt"
+  }
+  ["\(superLongPathPrefix)/someDir/otherDir"] = new File.Directory {}
+}
+
+renderer {
+  defaultFileInfo {
+    uid = 2 ** 21 as UInt
+    gid = 2 ** 21 as UInt
+    mtime = 1767225600000.123456 // 2026-01-01T00:00:00.123456Z
```

Comment:
Floating point math makes me nervous.

Maybe better to just have

```pkl
  /// File modification time in epoch nanoseconds.
  ///
  /// Truncated to whole seconds for UStar archives.
  /// Must be < 2**33 for UStar archives.
  mtime: Int
```

Reply:
Pax archives support encoding timestamps with fractional seconds and if we restricted this to integers it would be the only part of the pax standard we're explicitly not supporting.

Thoughts on this instead?
```pkl
mtime: UInt|String(matches(Regex(#"\d+(\.\d+)?"#)))
```

Reply:
That sounds good to me.

By the way, epoch nanos would still qualify as fractional seconds (with nanosecond precision)